mos_tests/rabbitmq_oslo/conftest.py

```diff
@@ -38,3 +44,57 @@ def fixt_kill_rpc_server_client(env):
         with node.ssh() as remote:
             cmd = 'pkill -f oslo_msg_check_'
             remote.execute(cmd)
+
+
+@pytest.yield_fixture
+def fixt_iptables_drop_reject(env, request):
+    """Choose one controller and apply IPTables rules"""
+    i_am_chosen_mark = '/tmp/i_am_chosen_mark.txt'
+
+    if request.param == 'drop':
+        table_modif = (
+            'iptables -I INPUT -p tcp -m tcp --dport 4369  -j DROP ; '
+            'iptables -I INPUT -p tcp -m tcp --dport 25672 -j DROP ; '
+            'iptables -I INPUT -p tcp -m tcp --dport 5672  -j DROP ; '
+            'iptables -I INPUT -p tcp -m tcp --dport 5673  -j DROP ; ')
+        table_modif_del = (
+            'iptables -D INPUT -p tcp -m tcp --dport 4369  -j DROP ; '
+            'iptables -D INPUT -p tcp -m tcp --dport 25672 -j DROP ; '
+            'iptables -D INPUT -p tcp -m tcp --dport 5672  -j DROP ; '
+            'iptables -D INPUT -p tcp -m tcp --dport 5673  -j DROP ; ')
+    elif request.param == 'reject':
+        table_modif = (
+            'iptables -I INPUT -p tcp -m tcp --dport 4369'
+            ' -j REJECT --reject-with icmp-host-prohibited ; '
+            'iptables -I INPUT -p tcp -m tcp --dport 25672'
+            ' -j REJECT --reject-with icmp-host-prohibited ; '
+            'iptables -I INPUT -p tcp -m tcp --dport 5672'
+            ' -j REJECT --reject-with icmp-host-prohibited ; '
+            'iptables -I INPUT -p tcp -m tcp --dport 5673'
+            ' -j REJECT --reject-with icmp-host-prohibited ; ')
+        table_modif_del = (
+            'iptables -D INPUT -p tcp -m tcp --dport 4369'
+            ' -j REJECT --reject-with icmp-host-prohibited ; '
+            'iptables -D INPUT -p tcp -m tcp --dport 25672'
+            ' -j REJECT --reject-with icmp-host-prohibited ; '
+            'iptables -D INPUT -p tcp -m tcp --dport 5672'
+            ' -j REJECT --reject-with icmp-host-prohibited ; '
+            'iptables -D INPUT -p tcp -m tcp --dport 5673'
+            ' -j REJECT --reject-with icmp-host-prohibited ; ')
+    else:
+        raise ValueError("Does'n know such param [{0}]!".format(request.param))
+
+    controller = random.choice(env.get_nodes_by_role('controller'))
+    logger.debug('Applying IPTables rules [{0}] to {1}'.format(
+        request.param, controller.data['ip']))
+    with controller.ssh() as remote:
+        # execute iptables DROP command
+        remote.check_call(table_modif)
+        # mark this controller
+        with remote.open(i_am_chosen_mark, 'w') as f:
```

Comment:
If you need to store, on which controller, more py.test wai is to determine fixture - `controller`.